rocky/rocky/views/ooi_report.py

```diff
@@ -198,47 +249,64 @@ def get(self, request, *args, **kwargs):
-        except HTTPError as e:
-            messages.error(self.request, _("Error generating report: {}").format(e))
-            return redirect(get_ooi_url("ooi_report", self.ooi.primary_key, self.organization.code))
-
-        # generate file name
-        report_name = "bevindingenrapport"
-        org_code = self.organization.code
-        ooi_id = self.ooi.primary_key
-        valid_time = self.get_observed_at().isoformat()
-        current_time = datetime.now(timezone.utc).isoformat()
-        language = "nl"
+    def _generate_report_file_name(self):
         report_file_name = "_".join(
             [
-                report_name,
-                language,
-                org_code,
-                ooi_id,
-                valid_time,
-                current_time,
+                "bevindingenrapport",
+                "nl",
+                self.organization.code,
+                self.ooi.primary_key,
+                self.get_observed_at().strftime(self.FILE_NAME_FRIENDLY_DATE_FORMAT),
+                datetime.now(timezone.utc).strftime(self.FILE_NAME_FRIENDLY_DATE_FORMAT),
             ]
         )
         # allow alphanumeric characters, dashes and underscores, replace rest with underscores
-        report_file_name = re.sub("[^0-9a-zA-Z-]", "_", report_file_name)
+        report_file_name = re.sub("[^\\w\\+-]", "_", report_file_name)
         report_file_name = f"{report_file_name}.pdf"
 
-        # open pdf as attachment
-        try:
-            return FileResponse(keiko_client.get_report(report_id), as_attachment=True, filename=report_file_name)
-        except (HTTPError, ReportNotFoundException):
-            messages.error(
-                self.request, _("Error generating report: Timeout reached. See Keiko logs for more information.")
-            )
-            return redirect(get_ooi_url("ooi_report", self.ooi.primary_key, self.organization.code))
+        return report_file_name
+
+
+@class_view_decorator(otp_required)
+class FindingReportPDFView(FindingListView, ReportPDFView):
+    paginate_by = None
+
+    def get(self, request, *args, **kwargs):
+        findings = [item["finding"] for item in super().get_queryset()]
+
+        return self.get_report(
+            "Organization",
+            self.organization.name,
+            {finding.primary_key: finding for finding in findings},  # Create store from finding list
+            self._generate_report_file_name(),
+            reverse("finding_list", kwargs={"organization_code": self.organization.code}),
+        )
+
+    def _generate_report_file_name(self):
+        file_name = "_".join(
+            [
+                "bevindingenrapport_nl",
+                self.organization.code,
+                datetime.now(timezone.utc).strftime(self.FILE_NAME_FRIENDLY_DATE_FORMAT),
+            ]
+        )
+
+        return f"{file_name}.pdf"
+
+
+class APIFindingReportPDFView(ViewSet, FindingReportPDFView):
```

Comment:
This is really ugly as hell. A viewset is something different than a standard Django class based view. You shouldn't create classes that inherent from both. It just doesn't make any sense wrt object oriented programming.

It will probably give a lot of issues, but it is hard to really understand all the implications of this which itself is a good reason we should not do this. One thing I can think of is error handling, if something goes wrong the view will add a message and redirect to the finding view. I don't think we should ever do something like that in an API.

Reply:
I agree, this was a quick attempt to get somewhat of an endpoint but setting reporting and the API up properly is definitely a better decision at this stage.